aten/src/ATen/native/cuda/AdaptiveMaxPooling3d.cu

```diff
@@ -28,12 +28,12 @@ namespace native {
 
 namespace {
 
-__device__ inline int start_index(int a, int b, int c) {
-  return (int)std::floor((float)(a * c) / b);
+__device__ inline int64_t start_index(int64_t a, int64_t b, int64_t c) {
+  return (a * c) / b;
```

Comment:
If I understand correctly, you could still overflow the a * c multiplication here, right? I wonder if we should do something like in https://stackoverflow.com/questions/5542037/avoiding-overflow-in-integer-multiplication-followed-by-division

Reply:
That is a good suggestion, I will implement it. Thanks!